main/src/ca/uwaterloo/flix/language/phase/typer/SchemaConstraintGen.scala

```diff
@@ -80,6 +80,28 @@ object SchemaConstraintGen {
     }
   }
 
+  def visitFixpointQueryWithProvenance(e: KindedAst.Expr.FixpointQueryWithProvenance)(implicit c: TypeContext, root: KindedAst.Root, flix: Flix): (Type, Type) = {
+    implicit val scope: Scope = c.getScope
+    e match {
+      case KindedAst.Expr.FixpointQueryWithProvenance(exps, select, withh, tvar, loc1) =>
+        val expsSchemaRows = exps.map(visitExp).map(_._1)
```

Comment:
You will want

```
(tpes, effs) = exps.map(visitExp).unzip
```

You then need to include the union of all effs in the result effect.